modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryEngine.java

```diff
@@ -36,4 +39,10 @@
      */
     List<FieldsQueryCursor<List<?>>> query(@Nullable QueryContext ctx, String schemaName, String qry, Object... params)
         throws IgniteSQLException;
+
+    /** */
+    Collection<? extends RunningQuery> runningQueries();
```

Comment:
If I'm not mistaken, the current approach to cancel query is to get instance of `RunningQuery` and invoke on it a `cancel` method. Unfortunately such an approach doesn't cover the case when current node doesn't run any fragment of the query of interest

Reply:
I see no problem with it. Looks like the described case is in the scope of `the KILL <query>` command implementation.

Reply:
then why do we need a limited API at all?

Reply:
I guess internal API should represent the modules abilities.
We can introduce the method `QueryEngine#kill(UUID queryId)` when this functionality will be available.